File: aspnet-core/src/Event_Planning_System.Core/Entities/Notification.cs
using Abp.Domain.Entities;
using Event_Planning_System.Authorization.Users;
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

public enum Notification_Type
{
    UpComming,
    AskForInvitation,
    Updated,
    Other
}

public enum Notification_Status
{
    Pending,
    Accepted,
    Rejected,
}

namespace Event_Planning_System.Enitities
{
    public class notification : Entity<int>
    {
        [Required(ErrorMessage = "Content is required.")]
        public string Content { get; set; }

        [Required(ErrorMessage = "Date is required.")]
        [DataType(DataType.DateTime, ErrorMessage = "Date must be a valid datetime.")]
        public DateTime Date { get; set; }

        [Required(ErrorMessage = "Notification type is required.")]
        public Notification_Type NType { get; set; }

        public bool isRead { get; set; }

        public Notification_Status status { get; set; }

        public bool IsReviewTaken { get; set; } = false;

        public long UserId { get; set; }

        [ForeignKey("UserId")]
        public virtual User User { get; set; }

        public long GuestId { get; set; }

        [ForeignKey("GuestId")]
        public virtual User Guest { get; set; }

        public int EventId { get; set; }

        [ForeignKey("EventId")]
        public virtual Event Event { get; set; }
    }
}

File: angular/src/shared/Models/Notification.ts
export class Notifications {
  constructor(
    public id: number,
    public content: string | undefined,
    public date: Date | undefined,
    public nType: number | undefined,
    public isRead: boolean | undefined,
    public isReviewTaken:boolean |undefined,
    public status: number | undefined,
    public guestId: number | undefined,
    public eventId: number | undefined, 
    public eventName?: string ,
    public startDate?: Date 
  ) { }
}
export class UpdateNotificationStatusDto {
  constructor(public id: number,
    public status: NotificationStatus) {  }
}
export class UpdateReminderStatusDto {
  constructor(public id: number,
    public isRead: boolean) {  }
}
export class UpdateReviewStatus{
  constructor(public id: number,
    public isReviewTaken: boolean) {  }
}

export enum NotificationStatus {
  Pending,
  Accepted,
  Rejected
}
